pkg/virtual/framework/forwardingregistry/rest.go

```diff
@@ -66,35 +94,49 @@ func NewStorage(ctx context.Context, resource schema.GroupVersionResource, apiEx
 		// TODO: what do we do on Destroy()?
 	}
 
-	delegate := DefaultDynamicDelegatedStoreFuncs(
+	store := DefaultDynamicDelegatedStoreFuncs(
 		factory, listFactory, destroyer,
 		strategy, tableConvertor,
 		resource, apiExportIdentityHash, categories,
 		dynamicClusterClient, []string{}, *patchConflictRetryBackoff, ctx.Done(),
 	)
-	store := wrapper(resource.GroupResource(), delegate)
+	if wrapper != nil {
+		wrapper.Decorate(resource.GroupResource(), store)
+	}
 
 	statusStrategy := customresource.NewStatusStrategy(strategy)
-	statusDelegate := DefaultDynamicDelegatedStoreFuncs(
+	statusStore := DefaultDynamicDelegatedStoreFuncs(
 		factory, listFactory, destroyer,
 		statusStrategy, tableConvertor,
 		resource, apiExportIdentityHash, categories,
 		dynamicClusterClient, []string{"status"}, *patchConflictRetryBackoff, ctx.Done(),
 	)
-	delegateUpdate := statusDelegate.UpdaterFunc
-	statusDelegate.UpdaterFunc = func(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	delegateUpdate := statusStore.UpdaterFunc
+	statusStore.UpdaterFunc = func(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
 		// We are explicitly setting forceAllowCreate to false in the call to the underlying storage because
 		// subresources should never allow create on update.
 		return delegateUpdate(ctx, name, objInfo, createValidation, updateValidation, false, options)
 	}
-	statusStore := wrapper(resource.GroupResource(), statusDelegate)
+	if wrapper != nil {
```

Comment:
You have an inherent nil check in the implementation of Decorate() - do we need it here?

Reply:
I would keep it - you never know if some other implementation omits the nil check.

Reply:
Sure thing